x-pack/plugins/security_solution/public/common/components/agents/agent_status/agent_response_action_status.tsx

```diff
@@ -0,0 +1,171 @@
+      hasPendingIsolate: boolean;
+      hasPendingUnIsolate: boolean;
+    }
+
+    const {
+      totalPending,
+      actionList,
+      wasReleasing,
+      wasIsolating,
+      hasMultipleActionTypesPending,
+      hasPendingIsolate,
+      hasPendingUnIsolate,
+    } = useMemo<PendingActionsState>(() => {
+      const list: Array<{ label: string; count: number }> = [];
+      let actionTotal = 0;
+      let actionTypesCount = 0;
+
+      Object.entries(pendingActions)
+        .sort()
+        .forEach(([actionName, actionCount]) => {
+          actionTotal += actionCount;
+          actionTypesCount += 1;
+
+          list.push({
+            count: actionCount,
+            label:
+              RESPONSE_ACTION_API_COMMAND_TO_CONSOLE_COMMAND_MAP[
+                actionName as ResponseActionsApiCommandNames
+              ] ?? actionName,
+          });
+        });
+
+      const pendingIsolate = pendingActions.isolate ?? 0;
+      const pendingUnIsolate = pendingActions.unisolate ?? 0;
+
+      return {
+        actionList: list,
+        totalPending: actionTotal,
+        wasReleasing: pendingIsolate === 0 && pendingUnIsolate > 0,
+        wasIsolating: pendingIsolate > 0 && pendingUnIsolate === 0,
+        hasMultipleActionTypesPending: actionTypesCount > 1,
+        hasPendingIsolate: pendingIsolate > 0,
+        hasPendingUnIsolate: pendingUnIsolate > 0,
+      };
+    }, [pendingActions]);
+
+    const badgeDisplayValue = useMemo(() => {
+      return hasPendingIsolate ? (
+        ISOLATING_LABEL
+      ) : hasPendingUnIsolate ? (
+        RELEASING_LABEL
+      ) : isIsolated ? (
+        ISOLATED_LABEL
+      ) : (
+        <FormattedMessage
+          id="xpack.securitySolution.agentStatus.agentAndActionsStatus.multiplePendingActions"
+          defaultMessage="{count} {count, plural, one {action} other {actions}} pending"
+          values={{
+            count: totalPending,
+          }}
+        />
+      );
+    }, [hasPendingIsolate, hasPendingUnIsolate, isIsolated, totalPending]);
+
+    const isolatedBadge = useMemo(() => {
+      return (
+        <EuiBadge color="hollow" data-test-subj={dataTestSubj}>
+          {ISOLATED_LABEL}
+        </EuiBadge>
+      );
+    }, [dataTestSubj]);
+
+    // If nothing is pending
+    if (totalPending === 0) {
+      // and host is either releasing and or currently released, then render nothing
+      if ((!wasIsolating && wasReleasing) || !isIsolated) {
+        return null;
+      }
+      // else host was isolating or is isolated, then show isolation badge
+      else if ((!isIsolated && wasIsolating && !wasReleasing) || isIsolated) {
+        return isolatedBadge;
+      }
+    }
+
+    // If there are different types of action pending
+    //    --OR--
+    // the only type of actions pending is NOT isolate/release,
+    // then show a summary with tooltip
+    if (hasMultipleActionTypesPending || (!hasPendingIsolate && !hasPendingUnIsolate)) {
```

Comment:
Q: what other kind of actions can we have here besides `(!hasPendingIsolate && !hasPendingUnIsolate)` ?

Reply:
We can have all other actions, but we prioritise showing `isolating`/`releasing` if there's such an action besides the other actions. This block excludes that as you can see in the comment above.